### DISCUSSION TITLE
Any examples of custom zones?

### DISCUSSION THREAD

**Discussion by ccfoo242** [moment/luxon#1270] [Q&A] [answered]
Hi all!

The docs mention custom zones and say this is an advanced topic. I've searched this repo and elsewhere and can't find any examples. I'm porting some C# code from my company to typescript and using luxon as a replacement for dotnet's DateTime. We have custom time zones for some of the energy markets we work with, so in our code we can use specify a "MISO" or "SPP" time zone.

Some of these are fixed zones. MISO uses US Eastern Standard Time year round. 
Some observe daylight time. SPP uses US Central Standard and Daylight. 

Is it possible to define these zones for use with luxon, and if so, how would I do that?

**Comment by icambron** [comment, marked as answer]
To make a custom zone, you write an implementation of the [Zone](https://github.com/moment/luxon/blob/master/src/zone.js) interface. Then you can pass instances of that object anywhere that takes zone for an argument. Luxon's [own zones](https://github.com/moment/luxon/tree/master/src/zones) do the same thing, with additional convenience that we've aliased strings to them so that, say, `setZone("utc")` is equivalent to `setZone(anInstanceOfUTCZone)`.

If you are OK passing a different kind of thing, you might not have to do much of anything, except something like:

```js
const MISO = FixedOffsetZone.instance(-4 * 60);
const SPP = "America/Chicago";
```

and then pass those values into anything that accepts zone as an argument.

However, if what you're after is being able to pass the *strings* "MISO" and "SPP" as args, then that isn't currently possible to customize.

I hope that helps!

**Reply by ccfoo242** [reply]
Ok, thanks! That helps. We currently have a static class with our custom timezones so I would pass TimeZones.MISO where a time zone was needed. I can do the same then and just have a static property MISO that returns FixedOffsetZone.instance(-4 * 60), for example.

My next task is handling DST but I will post a different question if I'm not able to reproduce the same behavior.
